api/resources_portal/views/organization_invitations.py

```diff
@@ -0,0 +1,52 @@
+from django.core.exceptions import PermissionDenied
+from rest_framework import serializers, status, viewsets
+from rest_framework.response import Response
+
+from resources_portal.models import Organization, OrganizationInvitation, User
+
+
+class OrganizationInvitationSerializer(serializers.ModelSerializer):
+    requester = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    request_reciever = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    organization = serializers.PrimaryKeyRelatedField(queryset=Organization.objects.all())
+
+    def validate(self, data):
+        if not data["request_reciever"].has_perm(
+            "add_members_and_manage_permissions", data["organization"]
```

Comment:
I don't think this quite covers the nuances of the permissions necessary for this route. The flow will be:
1. `requester` POSTs an invitation (permission here is a bit complex because anyone may make a `REQUEST`, but only someone with permission can create an `INVITE`).
2. `request_receiver` is the only one who may PUT to update an existing request by approving or declining it. (Although note that the `requester` is the only one who should be able to DELETE the invitation.)
3. Once an invitation has been approved or denied, it should no longer be mutable.

Reply:
Also I think this needs to be done in the view because you need to see who `request.user` is because you cannot assume `data['request_reciever']` is the user that made the HTTP request.

Reply:
Alright, I think the proper permissions are in place. Have a look at your leisure.